torch/distributed/distributed_c10d.py

```diff
@@ -346,7 +346,9 @@ def init_process_group(backend,
                                 Mutually exclusive with ``init_method``.
         timeout (timedelta, optional): Timeout for operations executed against
             the process group. Default value equals 30 minutes.
-            This is only applicable for the ``gloo`` backend.
+            This is applicable for the ``gloo`` backend. For ``nccl``, this is
+            applicable only if the environment variable NCCL_BLOCKING_WAIT is
```

Comment:
Q: Why does NCCL_BLOCKING_WAIT have to be set to 1 for the timeout to be applicable?

Reply:
Without `NCCL_BLOCKING_WAIT` work->wait() does not block and there isn't really something that the user is waiting on.

Reply:
` ``NCCL_BLOCKING_WAIT`` `